onnxruntime/test/autoep/library/example_kernel_plugin_ep/ep.cc

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "ep.h"
+
+#include <gsl/span>
+#include <array>
+#include <cassert>
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "ep_factory.h"
+#include "../plugin_ep_utils.h"
+
+KernelEp::KernelEp(KernelEpFactory& factory, const OrtLogger& logger)
+    : OrtEp{},  // explicitly call the struct ctor to ensure all optional values are default initialized
+      factory_{factory},
+      ort_api_{factory.GetOrtApi()},
+      ep_api_{factory.GetEpApi()},
+      name_{factory.GetEpName()},
+      logger_{logger} {
+  ort_version_supported = ORT_API_VERSION;  // set to the ORT version we were compiled with.
+
+  // Initialize the execution provider's function table
+  GetName = GetNameImpl;
+  GetCapability = GetCapabilityImpl;
+  GetKernelRegistry = GetKernelRegistryImpl;
+
+  // This is not a compiling EP, so don't need the following
+  Compile = nullptr;
+  ReleaseNodeComputeInfos = nullptr;
+
+  auto status = ort_api_.Logger_LogMessage(&logger_,
+                                           OrtLoggingLevel::ORT_LOGGING_LEVEL_INFO,
+                                           ("KernelEp has been created with name " + name_).c_str(),
+                                           ORT_FILE, __LINE__, __FUNCTION__);
+  // ignore status for now
+  (void)status;
```

Comment:
this might leak the `OrtStatus*`.

BTW, I have a fix for the other example plugin EPs in #26498:
https://github.com/microsoft/onnxruntime/pull/26498/commits/ccc659da66bac55d6c64dcdd5e5e9d9479447c42